src/main/java/seedu/address/logic/Messages.java

```diff
@@ -57,10 +57,15 @@ public static String format(Person person) {
                 .append(person.getAddress())
                 .append(";\n")
                 .append("Appointment: ")
-                .append(person.getAppointment().orElse(person.getAppointment().orElse(null)))
+                .append(person.getAppointment().map(Appointment::toString).orElse("N/A"))
                 .append("; Medical Histories: ");
 
-        person.getMedicalHistories().forEach(builder::append);
+        if (person.getMedicalHistories().isEmpty()) {
+            builder.append("N/A;");
+        } else {
+            person.getMedicalHistories().forEach(builder::append);
+            builder.append(";");
+        }
```

Comment:
Is the `;` necessary for these fields? The toString method of `MedicalHistory` already adds square brackets around its value, so the additional `;` append seems unnecessary.

Reply:
I think the ";" for the else condition might not be necessary, I will remove it, for the if condition, I think it is a good to have, reason is because I am appending a string directly instead of using the `toString` method of `MedicalHistory`. This ensures consistency with all the other fields.